prime-router/src/testIntegration/kotlin/datatests/MappingInventoryTests.kt

```diff
@@ -0,0 +1,79 @@
+package gov.cdc.prime.router.datatests
+
+import gov.cdc.prime.router.Report
+import gov.cdc.prime.router.cli.tests.CompareData
+import org.junit.jupiter.api.Test
+
+class MappingInventoryTests {
+    fun shouldTranslateLosslessly(inputFile: String, outputFile: String): CompareData.Result {
+        val outputSchemaPath = "metadata/hl7_mapping/ORU_R01/ORU_R01-test"
+
+        val testConfig = TranslationTests.TestConfig(
+            inputFile, Report.Format.FHIR, "", outputFile,
+            Report.Format.HL7, outputSchemaPath, true, null, null, null
+        )
+        return TranslationTests().FileConversionTest(
+            testConfig
+        ).runTest()
+    }
+
+    @Test
+    fun `XTN use code home`() {
```

Comment:
Does it make sense to tie these to the spreadsheet somehow? Should we at the very least have a comment at the top that points to the XTN page in the mapping inventory? Something like this?

```
/* Location of mapping inventory rules
XTN : PAGE HERE
PID: PAGE HERE
...
*/
```

Additionally, for each test, is it possible to tie each test, via a comment, to a row in the spreadsheet? This probably doesn't make sense since the spreadsheet can change? An idea to resolve this is we can maintain our own copy of the mapping inventory and manually sync with the official one once in a while. Part of the sync process could be writing new tests or updating the current tests? If this makes sense, this should probably be captured in the test plan. The test plan should discuss how we detect changes to the mapping inventory and then update our mappings and tests, if that's something we want to do (I think we probably should)? @mkalish any thoughts here?

Reply:
Storing them here makes sense to me and will make it easier when we need to fix something